src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Addresses.cs

```diff
@@ -12,6 +12,11 @@ internal static partial class StringParsingHelpers
         // and seperates the information about by each interface.
         internal static List<GatewayIPAddressInformation> ParseGatewayAddressesFromRouteFile(string filePath, string interfaceName)
         {
+            if (!File.Exists(filePath))
+            {
+                throw ExceptionHelper.CreateForInformationUnavailable();
+            }
```

Comment:
I had not thought about race conditions. These are files under /proc/net/*, so I just added these checks to make sure the information even existed there on your platform. I think since these are procfs files, there's not a risk of them being there one minute and going away the next.